legend-depot-core-services/src/main/java/org/finos/legend/depot/services/projects/ProjectsServiceImpl.java

```diff
@@ -301,17 +303,51 @@ public ProjectDependencyReport buildReportFromGraph(ProjectDependencyGraph depen
     }
 
     @Override
-    public List<ProjectDependencyWithPlatformVersions> getDependentProjects(String groupId, String artifactId, String versionId)
+    public List<ProjectDependencyWithPlatformVersions> getDependantProjects(String groupId, String artifactId, String versionId)
     {
+        return getDependantProjects(groupId, artifactId, versionId, false);
+    }
+
+    @Override
+    public List<ProjectDependencyWithPlatformVersions> getDependantProjects(String groupId, String artifactId, String versionId, boolean latestOnly)
+    {
+        List<StoreProjectVersionData> projectVersionDatas;
+        if (latestOnly)
```

Comment:
i dont think this implementation is correct .version id play a role here.. 
so if we are asked to get dependants for all versions of groupid/artifact.. how latest only modify the results?
if we get a request to get dependants for one gav .. how latestOnly modifies the results?

Reply:
Hello Elena, thanks for all of your comments. I should have added the new filter after the calculation instead of before the calculation. I've updated the code. Would you mind reviewing it again when you have time? Thank you.  🤩

Reply:
yep looks good..agree with the readabilty improvements suggested by Gunjan so please look into tthat

Reply:
sure I will resolve it today!